diff -up rsyslog-5.8.10/runtime/nsd_gtls.c.orig rsyslog-5.8.10/runtime/nsd_gtls.c
--- rsyslog-5.8.10/runtime/nsd_gtls.c.orig	2012-04-05 14:10:45.000000000 +0200
+++ rsyslog-5.8.10/runtime/nsd_gtls.c	2012-05-16 15:40:00.070328701 +0200
@@ -257,8 +257,8 @@ gtlsClientCertCallback(gnutls_session se
 static rsRetVal
 gtlsGetCertInfo(nsd_gtls_t *pThis, cstr_t **ppStr)
 {
-	char dn[128];
-	uchar lnBuf[256];
+	char dn[256];
+	uchar lnBuf[512];
 	size_t size;
 	unsigned int algo, bits;
 	time_t expiration_time, activation_time;
